# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mythtv/directfb-directresult.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- mythtv-0.21/libs/libmythtv/videoout_directfb.cpp.orig	2008-03-08 01:07:31.000000000 +0100
+++ mythtv-0.21/libs/libmythtv/videoout_directfb.cpp	2008-09-10 09:34:58.000000000 +0200
@@ -141,8 +141,7 @@
         if (DFB_OK != fberr)
         {
             VERBOSE(VB_IMPORTANT, LOC_ERR + "Failed to lock buffer" + ENOFB);
-            fberr = bufferSurface->Release(bufferSurface);
-            if (DFB_OK != fberr)
+            if (bufferSurface->Release(bufferSurface) != DR_OK)
             {
                 VERBOSE(VB_IMPORTANT, LOC_ERR +
                         "Failed to release buffer" + ENOFB);
@@ -198,8 +197,7 @@
                             "Failed to unlock buffer" + ENOFB);
                 }
 
-                fberr = surf->Release(surf);
-                if (DFB_OK != fberr)
+                if (surf->Release(surf) != DR_OK)
                 {
                     VERBOSE(VB_IMPORTANT, LOC_ERR +
                             "Failed to release buffer" + ENOFB);
@@ -244,8 +242,7 @@
                         "Failed to unlock buffer" + ENOFB);
             }
 
-            fberr = surf->Release(surf);
-            if (DFB_OK != fberr)
+            if (surf->Release(surf) != DR_OK)
             {
                 VERBOSE(VB_IMPORTANT, LOC_ERR +
                         "Failed to release buffer" + ENOFB);
@@ -1121,8 +1118,7 @@
     {
         VERBOSE(VB_IMPORTANT, LOC_WARN + "Failed to test config" + ENOFB);
 
-        fberr = vodata->videoLayer->Release(vodata->videoLayer);
-        if (DFB_OK != fberr)
+        if (vodata->videoLayer->Release(vodata->videoLayer) != DR_OK)
         {
             VERBOSE(VB_IMPORTANT, LOC_ERR +
                     "Failed to release display layer" + ENOFB);
